frontend/src/routes/Governance/discovered/grouping.ts

```diff
@@ -0,0 +1,205 @@
+  }
+
+  const getSourceText = (policySource: any, isExternal: boolean) => {
+    if (policySource.type) {
+      const channelType = (policySource.type && policySource.type.toLowerCase()) || ''
+      const normalizedChannelType: string = channelType === 'github' ? 'git' : channelType
+      return normalizedChannelType.charAt(0).toUpperCase() + normalizedChannelType.slice(1)
+    } else if (isExternal) {
+      return 'External'
+    } else {
+      return 'Local'
+    }
+  }
+
+  function getResourceMap(resources: any[]) {
+    const resourceMap: Record<string, any> = {}
+    resources.forEach((r: any) => {
+      const key = `${r.metadata?.namespace}/${r.metadata?.name}`
+      key && (resourceMap[key] = r)
+    })
+    return resourceMap
+  }
+
+  const resolveSource = (
+    annotations: { [key: string]: string },
+    helmReleases: any[],
+    channels: any[],
+    subscriptions: any[]
+  ) => {
+    const getAnnotations = (item: any) => item?.metadata?.annotations ?? {}
+    const getHostingSubscription = (annotations: any) =>
+      annotations['apps.open-cluster-management.io/hosting-subscription']
+    const parentAnnotations = annotations
+    let hostingSubscription = getHostingSubscription(parentAnnotations)
+    if (!hostingSubscription) {
+      // check if this policy was deployed by a Helm release
+      const releaseNamespace = parentAnnotations['meta.helm.sh/release-namespace']
+      const releaseName = parentAnnotations['meta.helm.sh/release-name']
+      if (releaseNamespace && releaseName) {
+        const helmReleaseMap = getResourceMap(helmReleases)
+        const helmRelease = helmReleaseMap[`${releaseNamespace}/${releaseName}`]
+        const helmReleaseAnnotations = getAnnotations(helmRelease)
+        hostingSubscription = getHostingSubscription(helmReleaseAnnotations)
+      }
+    }
+    if (hostingSubscription) {
+      const subscriptionMap = getResourceMap(subscriptions)
+      const channelMap = getResourceMap(channels)
+      const subscription = subscriptionMap[hostingSubscription]
+      const channel = channelMap[subscription?.spec.channel ?? '']
+      if (subscription && channel) {
+        return {
+          type: channel.spec.type ?? '',
+        }
+      }
+    }
+    return null
+  }
+
+  const getPolicySource = (policy: any, helmReleases: any[], channels: any[], subscriptions: any[]): ISourceType => {
+    const parentPolicy = getStringLabelValue(policy.label, 'policy.open-cluster-management.io/policy')
+    const parentNs = parentPolicy?.split('.')[0] ?? ''
+    const parentName = parentPolicy?.split('.')[1] ?? ''
+
+    if (parentNs && parentName) {
+      return {
+        type: 'Policy',
+        parentNs,
+        parentName,
+      }
+    }
+
+    const isExternal = policy._isExternal
+    let source: string | JSX.Element = 'Local'
+    if (isExternal) {
+      const policySource = resolveSource(policy.annotations ?? {}, helmReleases, channels, subscriptions)
```

Comment:
Is this right for the annotations? I thought it's `policy.annotation` and it's in the same format as `policy.label`.

Reply:
Great point!

Reply:
Good point!! I missed